ambry-api/src/main/java/com.github.ambry/store/StoreStats.java

```diff
@@ -40,10 +41,19 @@
   Pair<Long, Long> getValidSize(TimeRange timeRange) throws StoreException;
 
   /**
-   * Fetches stats for the corresponding {@link Store} as a {@link StatsSnapshot}.
+   * Fetches stats for the corresponding {@link Store} as a single {@link StatsSnapshot}.
    * @param referenceTimeInMs the reference time in ms until which deletes and expiration are relevant
    * @return a {@link StatsSnapshot} with relevant stats
    * @throws StoreException
    */
   StatsSnapshot getStatsSnapshot(long referenceTimeInMs) throws StoreException;
+
+  /**
+   * Fetches all types of stats for the corresponding {@link Store} as a map whose key is type name and value is
+   * {@link StatsSnapshot}.
+   * @param referenceTimeInMs the reference time in ms until which deletes and expiration are relevant
+   * @return a map of all types of {@link StatsSnapshot} with relevant stats
+   * @throws StoreException
+   */
+  Map<String, StatsSnapshot> getAllStatsSnapshots(long referenceTimeInMs) throws StoreException;
```

Comment:
because the two StatsSnapshots generated by store are not in final format of stats reports listed in enum. Actually, the two snapshots don't have partition info (see following formats)
```
   *        AccountSnapshot             |             ContainerSnapshot
   * --------------------------------------------------------------------------
   * {                                  |      {
   *   value: 1000,                     |        value: 1000,
   *   subMap: {                        |        subMap: {
   *     Account[1]:{                   |           Account[1]_Container[1]: {
   *       value: 200,                  |               value: 200,
   *       subMap: {                    |               subMap: null
   *         Container[1]:{             |           },
   *           value: 200,              |           Account[2]_Container[3]: {
   *           subMap: null             |               value: 800,
   *         }                          |               subMap: null
   *       }                            |           }
   *     },                             |        }
   *     Account[2]:{                   |      }
   *       value: 800,                  |
   *       subMap:{                     |
   *         Container[3]:{             |
   *           value: 800,              |
   *           subMap: null             |
   *         }                          |
   *       }                            |
   *     }                              |
   *   }                                |
   * }                                  |
   */
```

Reply:
hmm, after reconsideration, actually, it's a good idea to relate type of snapshot to type of report because each snapshot is only used for specific type of report. We can directly use report type instead of extra snapshot types to make it less ambiguous and more concise.